sw/device/lib/base/BUILD

```diff
@@ -0,0 +1,78 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+package(default_visibility = ["//visibility:public"])
+
+cc_library(
+    name = "base",
+    hdrs = [
+        "csr.h",
+        "csr_registers.h",
+        "macros.h",
+        "stdasm.h",
+        "mmio.h",
+        ],
+    deps = [
+        ":bitfield",
+        ":memory",
+        ":hardened",
+        ],
```

Comment:
These libraries can be added into `base` (add their hdrs and srcs).  They don't have any compile-time dependencies on preprocessor definitions like `MOCK_MMIO` and they're tiny and don't have any complex dependency graphs themselves.

Adding them into base will simplify their inclusion elsewhere: you can just write `//sw/device/lib/base` rather than having to additionally specify `.../base:bitfield`, etc.